src/WarcraftLegacies.Source/Quests/Goblin/QuestRatchet.cs

```diff
@@ -0,0 +1,61 @@
+﻿using System.Collections.Generic;
+using MacroTools.ControlPointSystem;
+using MacroTools.Extensions;
+using MacroTools.FactionSystem;
+using MacroTools.LegendSystem;
+using MacroTools.ObjectiveSystem.Objectives.ControlPointBased;
+using MacroTools.ObjectiveSystem.Objectives.FactionBased;
+using MacroTools.ObjectiveSystem.Objectives.LegendBased;
+using MacroTools.ObjectiveSystem.Objectives.UnitBased;
+using MacroTools.QuestSystem;
+using static War3Api.Common;
+
+
+namespace WarcraftLegacies.Source.Quests.Goblin
+{
+  /// <summary>
+  /// The Goblins can acquire Ratchet.
+  /// </summary>
+  public sealed class QuestRatchet : QuestData
+  {
+    private readonly List<unit> _rescueUnits;
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="QuestRatchet"/>.
+    /// </summary>
+    /// <param name="noggenfogger">Must be brought somewhere to complete the quest.</param>
+    public QuestRatchet(LegendaryHero noggenfogger) : base("Ratchet Port",
+      "The port of Ratchet would be a great expansion of our trade empire.",
+      @"ReplaceableTextures\CommandButtons\BTNGoblinShop2.blp")
+    {
+      AddObjective(new ObjectiveControlPoint(ControlPointManager.Instance.GetFromUnitType(Constants.UNIT_N0A6_RATCHET)));
+      AddObjective(new ObjectiveLegendInRect(noggenfogger, Regions.Ratchet_Unlock_1, "Ratchet"));
+      AddObjective(new ObjectiveSelfExists());
+      Required = true;
+      ResearchId = Constants.UPGRADE_VQ01_QUEST_COMPLETED_RATCHET_PORT;
+      _rescueUnits = Regions.Ratchet_Unlock_1.PrepareUnitsForRescue(RescuePreparationMode.HideNonStructures,
+        filterUnit => filterUnit.GetTypeId() != FourCC("ngme"));
+    }
+
+    /// <inheritdoc />
+    protected override string RewardFlavour => "With a port now established near Orgrimmar, we now have an outpost to support the Horde war effort in the North!";
```

Comment:
It's a huge improvement, but it still doesn't have that Ratchet feel; it just has "generic port" feel. I tend to look up lore features on wowpedia or wowwiki to get ideas of things to include. Maybe something like:
```suggestion
    protected override string RewardFlavour => "Baron Noggenfogger and his cronies have established the city of Ratchet, an ostensibly neutral port populated by smugglers and outcasts.";
```

Reply:
If we allow Gazlowe to found the city like I suggested, we'll also need to update the description to be based on whichever hero completed it. But I can show you how to do that after.

Reply:
Have updated it and referenced Gazlowe as the main founder, although as you mentioned it could often be Noggenfogger in game.

Reply:
Ok, here's where things get complicated. Now we need the reward popup to say either Noggenfogger or Gazlowe, depending on who completed it. `ObjectiveAnyUnitInRect` has a `CompletingUnit` property you can use for this exact purpose. Check out the `QuestTombOfSargeras` `RewardFlavour` for a solid example.

Reply:
Okay will have a look at that.

I added reference to the Bilgewater Cartel as I thought that was more specific than just smugglers and also the text of "support the horde in the north" is to kind of gode the player into moving north with their team, if they weren't already planning to do so.

Reply:
I don't think `RewardFlavour` should be used as a tool to explain gameplay concepts, because Legacies isn't just a competitive match - it's an exploration of the Warcraft universe. `RewardFlavour` in particular is, specifically, flavour. And canonically, Kezan had absolutely nothing to do with the Horde's war effort. It should contain things that are specific to Kezan as an entity in the Warcraft universe. And those things, to my reading on the Wiki page, are:
* That it was established by Gazlowe
* That it has many smugglers and outcasts
* That it is neutral